tests/integration/api/test_activation_with_credential.py

```diff
@@ -347,17 +199,49 @@ def test_is_activation_valid_with_run_job_template_and_no_token_no_credential(
     ) in message
 
 
+@pytest.mark.django_db
+def test_is_activation_valid_with_updated_credential(
+    default_activation: models.Activation,
+    preseed_credential_types,
+):
+    user_credential_type = create_user_credential_type()
+    credential = models.EdaCredential.objects.create(
+        name="eda-credential",
+        inputs={
+            "var_1": "adam",
+            "var_2": "secret",
+            "sasl_username": "adam",
+            "sasl_password": "secret",
+        },
+        credential_type_id=user_credential_type.id,
+    )
+
+    default_activation.eda_credentials.add(credential)
+
+    valid, _ = is_activation_valid(default_activation)
+    assert valid is True
+
+    default_activation.refresh_from_db()
+    extra_var = yaml.safe_load(default_activation.extra_var.extra_var)
+    assert extra_var["sasl_username"] == "adam"
+    assert extra_var["sasl_password"] == "secret"
+
+
 @pytest.mark.django_db
 def test_create_activation_with_eda_credential(
     client: APIClient,
     kafka_credential_type: models.CredentialType,
+    default_de: models.DecisionEnvironment,
+    default_rulebook: models.Rulebook,
+    default_extra_var: models.ExtraVar,
     preseed_credential_types,
 ):
-    fks = create_activation_related_data(TEST_EXTRA_VAR)
-    test_activation = TEST_ACTIVATION.copy()
-    test_activation["decision_environment_id"] = fks["decision_environment_id"]
-    test_activation["rulebook_id"] = fks["rulebook_id"]
-    test_activation["extra_var_id"] = fks["extra_var_id"]
+    test_activation = {
```

Comment:
We also have `activation_payload` fixture for POST activation, could that be used here?